content/learn/intermediate/data-types-2.md

```diff
@@ -1,178 +1,70 @@
-    \
-    In the above scenario, `.` is still given the successive elements of the slice. However, the variable `$value` also attains successive elements of the slice while the variable `$index` is passed the value of the successive indices of the slice (from `0` to `length -1`).\\
-8. `.StringSlice`\
-   Syntax : `<slice>.StringSlice <strict-flag (optional)>`\
-   \
-   This is a unique method which can be used to extract and return all string (or string representable) elements of a _template.Slice_ in form of a string slice (type _\[]string_). If none are _string_ an empty _\[]string_ slice is returned. It has an optional strict flag. f strict-flag is set to \_true \_it will return a _\[]string_ only if **all** elements are pure _string_, else `<no value>` is returned. This is also useful in a few cases where certain functions or methods only accept string slices and not a templates.Slice.\
-   \
-   Example :\
-   `{{$slice := cslice "This" 1 "Is" "Fun"}}{{$new := $slice.StringSlice}}`\
-   `{{joinStr "-" $new}}`\
-   \
-   Output :\
-   `This-Is-Fun`\
-   \
-   In the above snippet, initially a Slice is created with `cslice` called `$slice`. Then on invoking `.StringSlice` method on `$slice` , a new slice of type _\[]string_ containing only the _string_ datatype elements of _$slice_ is created and stored in `$new`. Then we using `joinStr` function to join the elements of the _\[]string_ slice into a single string with `-` as separator and the resultant string is output.
-
-{% hint style="info" %}
-Up until now we have been hearing about generic slices but it is interesting to note that we have already dealt with other types of slices than _templates.Slice_ datatype. Recall that `.CmdArgs` was defined as a slice of string arguments. Hence it is also a slice but of datatype _\[]String_.
-{% endhint %}
-
-{% hint style="info" %}
-Note that we have [already seen](https://learn.yagpdb.xyz/intermediate/control-flow-2#lets-go-deeper) the usage of the `seq` function in performing a repeated task a known number of times. However, did you know that seq actually also produces a slice. It is of type _\[]int_ and contains integers greater than equal to the first argument (start) upto one less than the second argument (stop).
-{% endhint %}
++++
+title = "Data Types 2"
+weight = 320
++++
+
+When you first started this course, you learned about [primitive data types](/learn/beginner/datatypes-1). In the
+previous chapter we name-dropped two more types: "slice", and "sdict" ("map"), both without much elaboration.
+In this chapter, we will explore these data types in more detail.
+
+## Slices
+
+Think of a slice as a list of items, that you can add to or remove from. You might have already heard of an array, which
+is a fixed-size list of items. A slice is like an array, but it can grow or shrink in size. This makes slices more
+powerful than a simple array.
+
+In custom commands, we can **c**reate a **slice** with the `cslice` function, like so:
```

Comment:
we can create a slice by providing the elements in order to the `cslice` function